packages/dds/tree/src/shared-tree-core/transaction.ts

```diff
@@ -163,3 +172,141 @@ export class TransactionStack implements Transactor, IDisposable {
+	 * @remarks This editor can safely be held on to across transaction boundaries, as it will properly delegate to the correct branch.
+	 * In contrast, it is not safe to hold onto e.g. `activeBranch.editor` across transaction boundaries, since the active branch may change.
+	 */
+	public readonly activeBranchEditor = new Proxy<TEditor>({} as unknown as TEditor, {
+		get: (_, p, receiver) => {
+			return Reflect.get(this.activeBranch.editor, p, receiver);
+		},
+	});
+
+	/**
+	 * Get the "active branch" for this transactor - either the transaction branch if a transaction is in progress, or the original branch otherwise.
+	 */
+	public get activeBranch(): SharedTreeBranch<TEditor, TChange> {
+		return this.#transactionBranch ?? this.branch;
+	}
+
+	/**
+	 * Provides events for changes in the {@link SquashingTransactionStack.activeBranch | active branch}.
+	 * @remarks When the active branch changes, the listeners for these events will automatically be transferred to the new active branch.
+	 * In contrast, binding an event to the {@link SquashingTransactionStack.activeBranch | active branch} directly will not automatically transfer the listener when the active branch changes.
+	 */
+	public get activeBranchEvents(): Listenable<SharedTreeBranchEvents<TEditor, TChange>> {
+		const off = (
+			eventName: keyof SharedTreeBranchEvents<TEditor, TChange>,
+			listener: SharedTreeBranchEvents<TEditor, TChange>[typeof eventName],
+		): void => {
+			this.activeBranch.events.off(eventName, listener);
+			const listeners = this.#activeBranchEvents.get(eventName);
+			if (listeners?.delete(listener) === true && listeners.size === 0) {
+				this.#activeBranchEvents.delete(eventName);
+			}
+		};
+
+		return {
+			on: (eventName, listener) => {
+				const listeners = getOrCreate(this.#activeBranchEvents, eventName, () => new Set());
+				listeners.add(listener);
+				this.activeBranch.events.on(eventName, listener);
+				return () => off(eventName, listener);
+			},
+			off,
+		};
+	}
+	readonly #activeBranchEvents = new Map<
+		keyof SharedTreeBranchEvents<TEditor, TChange>,
+		Set<
+			SharedTreeBranchEvents<TEditor, TChange>[keyof SharedTreeBranchEvents<TEditor, TChange>]
+		>
+	>();
+
+	/**
+	 * Construct a new {@link SquashingTransactionStack}.
+	 * @param branch - The {@link SquashingTransactionStack.branch | branch} that will be forked off of when a transaction begins.
+	 * @param squash - Called once when the outer-most transaction is committed to produce a single squashed change from the transaction's commits.
+	 * The change will be applied to the original {@link SquashingTransactionStack.branch | branch}.
+	 * @param onPush - {@link OnPush | A function} that will be called when a transaction is pushed to the {@link TransactionStack | stack}.
+	 */
+	public constructor(
+		/** The branch that will be forked off of when a transaction begins */
```

Comment:
Is this doc comment needed give the one above? (Same for `squash`)

Reply:
No, thanks.